src/js/components/Layer/StyledLayer.js

```diff
@@ -743,7 +743,7 @@ const desktopContainerStyle = css`
       'right',
     )}px)`};
   ${props =>
-    props.plain || (props.full === true && props.margin === 'none')
+    props.plain || (props.full && props.margin === 'none')
```

Comment:
**Note:** Chatted through this with @taysea, but leaving these thoughts for others to consider.

How does this work if I want my Layer to positioned incoming from the position="right", full="vertical", but I want my corners to be `24px 0 0 24px`? Am I able to do that with this approach? Seems like all corners get set to `0` but I may be missing something.

**Desire**
![Screen Shot 2021-01-26 at 2 55 03 PM](https://user-images.githubusercontent.com/1756948/105915205-ac25a180-5fec-11eb-8b11-ec0498ed4691.png)

--> **Outcome of our conversation:** Decided that this implementation as is advances the current asks and we can wait for the community to express the need for "smart corners" when in full modes.